File: Python 3/CodingProblems/longestWord.py
"""
 Have the function LongestWord(sen) take the sen parameter being passed and return the largest word in the string. 
 If there are two or more words that are the same length, return the first word from the string with that length.
 Ignore punctuation and assume sen will not be empty.
"""

# Big O          Complexity: O(n)       Space: O(n)

def LongestWord(sen):

  # With re library
  import re
  # code goes here
  sen = re.sub("[^0-9a-zA-Z]+", " ", sen)
  tempSen = sen.split(" ")

  largest = ""
  for word in tempSen:
    if len(word) > len(largest):
      largest = word
  return largest

# keep this function call here 
print(LongestWord(input()))


# Without Re library
def LongestWord(sen):

  newSen = ""
  for letter in sen:
    if letter.isalnum() == True or letter == " ":
      newSen += letter

  newSen = newSen.split(" ")
  largest = ""

  for word in newSen:
    if len(word) > len(largest):
      largest = word
  return largest 

# keep this function call here 
print(LongestWord(input()))

File: JavaScript/CodingProblems/longestWord.js
/*
* Have the function LongestWord(sen) take the sen parameter being passed and return the largest word in the string. 
* If there are two or more words that are the same length, return the first word from the string with that length.
* Ignore punctuation and assume sen will not be empty.
*/

// Big O          Complexity: O(n)       Space: O(n)

function LongestWord(sen) { 
  let tempSen = sen.replace(/[^a-zA-Z0-9 ]/g, '');

  let arr = tempSen.split(" ");

  let maxTime = arr[0];
  for(let i = 1; i < arr.length; i++){
    let tempMax = arr[i];
    if (tempMax.length > maxTime.length ){
      maxTime = tempMax
    }
  }
  return maxTime; 
}
   
// keep this function call here 
console.log(LongestWord(readline()));
